crates/ibc/src/core/ics04_channel/handler/acknowledgement.rs

```diff
@@ -24,35 +23,21 @@ where
     let chan_end_path_on_a = ChannelEndPath::new(&packet.port_id_on_a, &packet.chan_id_on_a);
     let chan_end_on_a = ctx_a.channel_end(&chan_end_path_on_a)?;
 
-    if !chan_end_on_a.state_matches(&State::Open) {
-        return Err(PacketError::ChannelClosed {
-            channel_id: packet.chan_id_on_a.clone(),
-        }
-        .into());
-    }
+    // Checks the channel end not be `Closed`.
+    // This allows for optimistic packet processing before a channel opens
+    chan_end_on_a.verify_not_closed()?;
```

Comment:
this is different: previously, we make sure that the connection is OPEN; here, you check that it is not CLOSED.

We care about OPEN; INIT, TRY_OPEN, etc. are all invalid states.

Note that

```rust
    chan_end_on_a.verify_state_matches(&ChannelState::Open)?;
```

doesn't work either because it doesn't return the right error

Reply:
Ah, correct. [7bec947](https://github.com/cosmos/ibc-rs/pull/650/commits/7bec947f9faa4dbaefcf1e11a725236326cbd300)
But I think `chan_end_on_a.verify_state_matches(&ChannelState::Open)?;` works. The relevant error message got more generalized, it suits for this check as well, IIUC?